cmd/schema/generate.go

```diff
@@ -1,59 +1,49 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"os"
-	"reflect"
-	"strings"
+	"path/filepath"
 
-	"github.com/alecthomas/jsonschema"
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io"
 	v1alpha5 "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
-	"github.com/weaveworks/eksctl/pkg/utils/ipnet"
+	schemapkg "github.com/weaveworks/eksctl/pkg/schema"
 )
 
-func typeMapper(t reflect.Type) *jsonschema.Type {
-	if t == reflect.TypeOf(&ipnet.IPNet{}) {
-		return &jsonschema.Type{Type: "string"}
-	}
-	return nil
-}
-
 func main() {
 	if len(os.Args) != 2 {
 		panic("expected one argument with the output file")
 	}
 	outputFile := os.Args[1]
 
-	var document strings.Builder
-	reflector := jsonschema.Reflector{
-		RequiredFromJSONSchemaTags: true,
-		TypeMapper:                 typeMapper,
+	input := filepath.Join("../../../..", "pkg", "apis", "eksctl.io")
+	schema, err := schemapkg.GenerateSchema(input, "v1alpha5", "ClusterConfig", false)
+	if err != nil {
+		panic(err)
 	}
-	schema := reflector.Reflect(&v1alpha5.ClusterConfig{})
-	// We have to manually add examples here, because we can't tag `TypeMeta`
-	// from the k8s package
-	if kind, ok := schema.Definitions["ClusterConfig"].Properties.Get("kind"); ok {
-		t := kind.(*jsonschema.Type)
-		t.Examples = []interface{}{"ClusterConfig"}
+
+	// We add some examples and blacklist some descriptions
+	if t, ok := schema.Definitions["ClusterConfig"].Properties["kind"]; ok {
```

Comment:
This might also be possible to override using the `// +jsonschema` directives on fields.

Reply:
Note this code is exactly the same as beforehand when we used the `alecthomas/jsonschema` package.